src/sdam/monitor.rs

```diff
@@ -74,6 +88,7 @@ fn monitor_server_check(
     conn: &mut Connection,
     mut server_type: ServerType,
     server: &Weak<Server>,
+    make_connection: impl Fn() -> Connection,
```

Comment:
a docstring explaining why this lambda is necessary might be useful.

Reply:
We need to make a new connection to replace the old one, which requires knowing the address, the connect timeout, and the TLS options. Abstracting them into a lambda that could be passed along rather than passing along three separate arguments that were used for only one thing felt cleaner.

Reply:
I meant leave that comment in the code itself. I understand why it's necessary, but future readers of the codebase may not, or we may forget the exact reasoning in the future.

Reply:
Okay, will do.